torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -59,8 +59,18 @@ void PropagateShapeOnNode(Node * node) {
     // to get a quick and dirty propagation
     case kneg: {
       node->output()->setType(types[0]->contiguous());
-      break;
-    } default: {
+    } break;
+    case kReplaceIfUndef: {
+      node->output()->setType(types[0]->shared_from_this());
```

Comment:
If types[0] has a type, then it is not defined, and the type will get set to types[0] because that will be the value propagated. If its type is _not_ defined, then unification is an undefined type.

I will add a comment. I believe this does not get reached by our actual pathways, but I put it here to keep it from faulting if we re-arrange how these are used.